test/scripts/run-gateway-conformance.sh

```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env bash
+
+# Copyright Project Contour Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.  You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+set -o pipefail
+set -o errexit
+set -o nounset
+
+readonly KUBECTL=${KUBECTL:-kubectl}
+readonly IMG=${CONTOUR_E2E_IMAGE:-ghcr.io/projectcontour/contour:main}
+
+echo "Using Contour image: ${IMG}"
+
+${KUBECTL} apply -f examples/gateway-provisioner/00-common.yaml
+${KUBECTL} apply -f examples/gateway-provisioner/01-roles.yaml
+${KUBECTL} apply -f examples/gateway-provisioner/02-rolebindings.yaml
+${KUBECTL} apply -f <(sed "s|ghcr.io/projectcontour/contour:main|$IMG|g; s|imagePullPolicy: Always|imagePullPolicy: IfNotPresent|g" examples/gateway-provisioner/03-gateway-provisioner.yaml)
+
+${KUBECTL} apply -f - <<EOF
+kind: GatewayClass
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: contour
+spec:
+  controllerName: projectcontour.io/gateway-provisioner
+EOF
+
+cd $(mktemp -d)
+git clone https://github.com/kubernetes-sigs/gateway-api
```

Comment:
My thinking was to let them track `HEAD` since the upstream tests are very much a WIP (there are only ~6 right now, many more planned to be added), so we keep up with the latest automatically, but it does have the downside that they could start failing at any moment. I was planning to leave them as not required so they wouldn't block PRs in that case.

Reply:
Can always circle back here and revise if the current approach becomes too noisy, or once the tests stabilize.

Reply:
🤙🏽